src/model/formulations/base/unit.jl

```diff
@@ -44,6 +44,7 @@ _is_initially_on(g::Unit)::Float64 = (g.initial_status > 0 ? 1.0 : 0.0)
 
 function _add_reserve_vars!(model::JuMP.Model, g::Unit)::Nothing
     reserve = _init(model, :reserve)
+    reserve_shortfall = _init(model, :reserve_shortfall)
```

Comment:
Are we creating the `model[:reserve_shortfall]` variables anywhere? Note that `_init` simply create an empty dictionary. If we are really missing these variables, this also indicates that we don't have enough tests for this feature, since they didn't catch this trivial issue.

Reply:
I am not sure I follow, probably because I did not quite understand the flow with `_init`. The `model[:reserve_shortfall]` variables are used in base/system.jl.

Reply:
We should have `reserve_shortfall[t] = @variable(model, lower_bound = 0)` somewhere, but I don't see it in this PR. Without this line, the code in `base/system.jl` will fail, since `reserve_shortfall` is just an empty dictionary.

Reply:
Ah, right. Added in 97b8611.

Reply:
Nice. I have just added a test that fails if you remove that line.